src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java

```diff
@@ -781,6 +797,48 @@ private static void addPlaces(Leg leg, List<State> states, List<Edge> edges, boo
         }
     }
 
+    private static List<TransitTripStage> generateTransitTripStages(List<State> states) {
+        List<TransitTripStage> transitTripStages = new ArrayList<>();
+
+        int currentTripTime = 1;
+        int firstTransitFareStartsAt = -1;
+        Stop currentStop;
```

Comment:
Please move declaration of those two variables `currentStop` and `distance` to their definitions (they are used as inner variables inside `if` blocks and not shared)